syntax = "proto2";

package PlacerProto;

enum eOrientation {
    eNorth = 0;
    eWest  = 1;
    eSouth = 2;
    eEast  = 3;
}

enum eDirection {
    eInput          = 0;
    eOutput         = 1;
    eBiDirectional  = 2;
    eUnknown        = 3;
}

message Macro {
  required string name = 1;
  required string id   = 2;
  required int32  width = 3;
  required int32  height = 4;
  optional eOrientation orientation = 5 [default = eNorth];
  optional int32  lx = 6;
  optional int32  ly = 7;
}

message Terminal {
    required string name    = 1;
    required eDirection dir = 2;
    optional int32 pos_x    = 3;
    optional int32 pos_y    = 4;

}

message Layout {
    required int32 lx = 1;
    required int32 ly = 2;
    optional int32 ux = 3;
    optional int32 uy = 4;
}
 
message MacroCircuit {
    required Layout l   = 1;
    repeated Macro m    = 2;
    repeated Terminal t = 3;
}
